data-prepper-plugins/saas-source-plugins/crowdstrike-source/src/main/java/org/opensearch/dataprepper/plugins/source/crowdstrike/CrowdStrikeSource.java

```diff
@@ -0,0 +1,67 @@
+package org.opensearch.dataprepper.plugins.source.crowdstrike;
+
+
+import com.google.common.annotations.Beta;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.acknowledgements.AcknowledgementSetManager;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.plugin.PluginFactory;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+import org.opensearch.dataprepper.plugins.source.crowdstrike.rest.CrowdStrikeAuthClient;
+import org.opensearch.dataprepper.plugins.source.source_crawler.CrawlerApplicationContextMarker;
+import org.opensearch.dataprepper.plugins.source.source_crawler.base.Crawler;
+import org.opensearch.dataprepper.plugins.source.source_crawler.base.CrawlerSourcePlugin;
+import org.opensearch.dataprepper.plugins.source.source_crawler.base.PluginExecutorServiceProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.opensearch.dataprepper.plugins.source.crowdstrike.utils.Constants.PLUGIN_NAME;
+
+
+/**
+ * CrowdStrike connector entry point.
+ * 🚧 Work in progress — under active development.
+ * Not ready for production use.
+ */
+@Beta
+@DataPrepperPlugin(name = PLUGIN_NAME,
+        pluginType = Source.class,
+        pluginConfigurationType = CrowdStrikeSourceConfig.class,
+        packagesToScan = {CrawlerApplicationContextMarker.class, CrowdStrikeSource.class}
+)
+public class CrowdStrikeSource extends CrawlerSourcePlugin {
+
+    private static final Logger log = LoggerFactory.getLogger(CrowdStrikeSource.class);
+    private final CrowdStrikeSourceConfig sourceConfig;
+    private final CrowdStrikeAuthClient authClient;
+
+    @DataPrepperPluginConstructor
+    public CrowdStrikeSource(final CrowdStrikeSourceConfig sourceConfig,
+                             final PluginMetrics pluginMetrics,
+                             final PluginFactory pluginFactory,
+                             final AcknowledgementSetManager acknowledgementSetManager,
+                             final CrowdStrikeAuthClient authClient,
+                             Crawler crawler, PluginExecutorServiceProvider executorServiceProvider) {
+        super(PLUGIN_NAME, pluginMetrics, sourceConfig, pluginFactory, acknowledgementSetManager, crawler, executorServiceProvider);
+        log.info("Creating CrowdStrike Source Plugin");
+        this.sourceConfig = sourceConfig;
+        this.authClient = authClient;
+    }
+
+    @Override
+    public void start(Buffer<Record<Event>> buffer) {
+        log.info("Starting CrowdStrike Source Plugin...");
+        authClient.initCredentials();
+        // super.start(buffer);
```

Comment:
I'll uncomment this code in follow up PR when CrowdStrikeService and Client classes are implemented. cc: @san81 @engechas